### DISCUSSION TITLE
mini.operators is my new favorite

### DISCUSSION THREAD

**Discussion by 9mm** [nvim-mini/mini.nvim#625] [General]
I think I asked about this in a recent issue (swapping arguments)

Since then I really looked at docs for `mini.operators`...

`gmVab` is life changing..................

On top of this, I also realized I can use `b` and `f` for `vab...ab..ab..ab` instead of `va{a[a{`, or `vaq`, etc.

`gx` for swapping stuff I don't use AS much but still have used it 2 or 3 times this week and its awesome. I use `gm` like 40 times a day though..... 

Both of these are just so insanely good I can't even.......

I wish you would stream/record on youtube you using your own config to write these plugins. Even if you don't talk or edit them, and it's just raw typing, I bet theres so many more gems given that you're the one who's probably most familiar with them.

Do you have any absolute favorites, maybe top 3?

**Comment by echasnovski** [comment]
Thanks for kind words!

> Do you have any absolute favorites, maybe top 3?

What exactly? Frequently used key combinations for editing?

Off the top of my head:
- I found myself quite frequently using `gmm` and `grr` to multiply and replace current line.
- `gr` in combination with `in)` to replace something in the next `()` (even parenthesis is empty) is quite nice. In general "next"/"prev" variants of textobjects in 'mini.ai' are very useful together with dot-repeat. Like `sr` - `n{` - `{` will ensure that `{}` brackets have padding, and it is dot-repeatable.
- Using `sdf` from 'mini.surround' feels nice with regular QWERTY keyboard.

**Reply by 9mm** [reply]
Me after learning about `grr`. Now thaz what I'm talking about!!!

![](https://media2.giphy.com/media/pHMtogPBHC7Cg/giphy.gif)

Need to reread on your others like `sdf`
